core/templates/pages/story-editor-page/navbar/story-editor-navbar.directive.ts

```diff
@@ -75,32 +81,37 @@ angular.module('oppia').directive('storyEditorNavbar', [
             $scope.validationIssues = $scope.story.validate();
             var nodes = $scope.story.getStoryContents().getNodes();
             var explorationIds = [];
-            for (var i = 0; i < nodes.length; i++) {
-              if (
-                nodes[i].getExplorationId() !== null &&
-                nodes[i].getExplorationId() !== '') {
-                explorationIds.push(nodes[i].getExplorationId());
-              } else {
-                $scope.validationIssues.push(
-                  'Some chapters don\'t have exploration IDs provided.');
-              }
-            }
 
-            if (StoryEditorStateService.getExpIdsChanged()) {
-              StoryEditorStateService.resetExpIdsChanged();
-              EditableStoryBackendApiService.validateExplorations(
-                $scope.story.getId(), explorationIds
-              ).then(function(validationIssues) {
-                $scope.explorationValidationIssues = validationIssues;
-                $scope.validationIssues =
-                  $scope.validationIssues.concat(
-                    $scope.explorationValidationIssues);
-              });
-            } else {
-              $scope.validationIssues =
-                $scope.validationIssues.concat(
-                  $scope.explorationValidationIssues);
+            if (
+              StoryEditorStateService.getExpIdsChanged() ||
+              $scope.forceValidateExplorations) {
+              $scope.explorationValidationIssues = [];
+              for (var i = 0; i < nodes.length; i++) {
+                if (
+                  nodes[i].getExplorationId() !== null &&
```

Comment:
I'm kinda worried about all this "is null or is empty string" constructs in the codebase. I think that an exploration ID should be null or nonempty.

I suggest adding a trashcan icon next to the exploration ID field to allow the user to delete it (it sets the field to null). Also, if the user clears the field, then, on-blur / on-save (whichever is appropriate), change the value to null. That way, you only check the null case and don't have to ever handle ''.

Reply:
Done, added the separate delete icon only visible if story is not  published.